securedrop/manage.py

```diff
@@ -23,7 +25,37 @@
 log = logging.getLogger(__name__)
 
 
-def reset():  # pragma: no cover
+def sh(command, input=None):
+    log.debug(":sh: " + command)
+    if input is None:
+        stdin = None
+    else:
+        stdin = subprocess.PIPE
+    proc = subprocess.Popen(
+        args=command,
+        stdin=stdin,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        shell=True,
+        bufsize=1)
+    if stdin is not None:
+        proc.stdin.write(input)
+        proc.stdin.close()
+    lines = []
+    with proc.stdout:
+        for line in iter(proc.stdout.readline, b''):
+            line = line.decode('utf-8')
+            lines.append(line)
+            log.debug(line.strip().encode('ascii', 'ignore'))
```

Comment:
because logging needs ascii so we need to convert lines that may contain non-ascii into something logging can accept. It could be an inconvenience if the non-ascii characters being ignored give a clue about an error. But I can't imagine any scenario where this actually is the case so I think we're good. It is possible I'm missing something though, do you have an idea in mind ?